### ISSUE TITLE
Problem when using non-default block_start_string, et. al. settings on Jinja

### ISSUE THREAD

**Issue by davidkhess** [inveniosoftware/flask-sitemap#42]
I have a Flask stack that I reuse with a Vue frontend. Since most of my HTML templates don't use Jinja syntax much but have a lot of Vue syntax, I've changed the default delimiter options (http://svn.python.org/projects/external/Jinja-1.2/docs/html/altsyntax.html) to be based on `[` instead.

However, this is a problem because this extension has embedded Jinja-based XML templates that use the default delimiters based on `{`. As a result, no actual Jinja rendering of the XML template occurs.

I don't think this is really a bug in flask-sitemap, but do you have any suggestions on how to deal with this? I've been source code diving on Flask and Jinja and Flask seems to only support one Jinja environment for the entire app. Jinja's Environment has a nice `.overlay()` method that I think would be useful in this case, but I can't find any way of using this without monkey-patching Flask which I would like to avoid.

Am I missing some more obvious way of dealing with this situation? Any other suggestions of where to look or what to try?

**Comment by jirikuncar** [comment]
@davidkhess you can try add a listener to `before_render_template` signal and modify context settings.

**Comment by davidkhess** [comment]
I did experiment with that sort of approach but didn’t have any success with it because of how templates are compiled in advance. There’s no hooks at that point of the life cycle. 

After chatting with the Jinja folks, the only reasonable solution we could come up with was to copy the templates to the app folder itself with the same path. In those copies I changed the delimiters. 

The template lookup finds those first now and everything works. Not pretty but it seemed to be the least ugly approach.
